crates/protocol/interop/src/safety.rs

```diff
@@ -19,6 +21,27 @@ pub enum SafetyLevel {
     Invalid,
 }
 
+impl FromStr for SafetyLevel {
+    type Err = SafetyLevelParseError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().as_str() {
+            "finalized" => Ok(Self::Finalized),
+            "safe" => Ok(Self::Safe),
+            "local_safe" | "localsafe" => Ok(Self::LocalSafe),
+            "cross_unsafe" | "crossunsafe" => Ok(Self::CrossUnsafe),
+            "unsafe" => Ok(Self::Unsafe),
+            "invalid" => Ok(Self::Invalid),
+            _ => Err(SafetyLevelParseError),
+        }
+    }
+}
+
+/// Error when parsing SafetyLevel from string.
+#[derive(Error, Debug)]
+#[error("Invalid SafetyLevel")]
+pub struct SafetyLevelParseError;
```

Comment:
for better error feedback the error should include the unknown safety level.

for example if this is parser from incorrect user input, then it's nice to include the invalid input value